librad/src/git.rs

```diff
@@ -42,72 +43,39 @@ const CONTRIBUTOR_METADATA_FILE: &str = "contributor.json";
 
 const RAD_REMOTE_NAME: &str = "rad";
 
-#[derive(Debug, Fail)]
+#[derive(Debug, Error)]
 pub enum Error {
-    #[fail(display = "Invalid PGP key: missing UserID packet")]
+    #[error("Invalid PGP key: missing UserID packet")]
     MissingPgpUserId,
 
-    #[fail(display = "Invalid PGP key: missing address")]
+    #[error("Invalid PGP key: missing address")]
     MissingPgpAddr,
 
-    #[fail(display = "Project already exists")]
+    #[error("Project already exists")]
     ProjectExists,
 
-    #[fail(display = "Project not found")]
+    #[error("Project not found")]
     NoSuchProject,
 
-    #[fail(
-        display = "Branch {} specified as default_branch does not exist in the source repo",
-        0
-    )]
-    MissingDefaultBranch(String, #[fail(cause)] git2::Error),
+    #[error("Branch {0} specified as default_branch does not exist in the source repo")]
+    MissingDefaultBranch(String, #[source] git2::Error),
 
-    #[fail(display = "{}", 0)]
-    Libgit(git2::Error),
+    #[error("Git error: {0:?}")]
+    Libgit(#[from] git2::Error),
 
-    #[fail(display = "{}", 0)]
-    Io(io::Error),
+    #[error(transparent)]
+    Io(#[from] io::Error),
 
-    #[fail(display = "{}", 0)]
-    Serde(serde_json::error::Error),
+    #[error(transparent)]
+    Serde(#[from] serde_json::error::Error),
 
-    #[fail(display = "{}", 0)]
-    Pgp(keys::pgp::Error),
+    #[error(transparent)]
+    Pgp(#[from] keys::pgp::Error),
 
-    #[fail(display = "{:?}", 0)]
+    #[error("Surf error: {0:?}")]
     Surf(surf::error::Error),
```

Comment:
Nb. `surf::error::Error` is not `std::error::Error`, so we can't use `#[from]` or `#[source]` here. radicle-dev/radicle-surf#87 should fix this